src/index.vue

```diff
@@ -683,26 +685,34 @@ const VueSelect = {
     }))
     provide('dataAttrs', dataAttrs)
 
-    const innerPlaceholder = computed(() => {
-      const selectedOptions = optionsWithInfo.value.filter(option => option.selected).filter(option => !option.group)
+    const selectedOptions = computed(() => {
+      return optionsWithInfo.value.filter(option => option.selected).filter(option => !option.group)
+    })
 
+    const innerPlaceholder = computed(() => {
       if (props.multiple) {
-        if (selectedOptions.length === 0) {
+        if (selectedOptions.value.length === 0) {
           return props.placeholder
-        } else if (selectedOptions.length === 1) {
+        } else if (selectedOptions.value.length === 1) {
           return '1 option selected'
         } else {
-          return selectedOptions.length + ' options selected'
+          return selectedOptions.value.length + ' options selected'
         }
       } else {
-        if (selectedOptions.length === 0) {
+        if (selectedOptions.value.length === 0) {
           return props.placeholder
         } else {
-          return selectedOptions[0].label + ''
+          return selectedOptions.value[0].label + ''
         }
       }
     })
 
+    const selected = computed(() => {
+      const selected = selectedOptions.value.map(option => option.originalOption)
+      if (props.multiple) return selected
+      return selected[0]
```

Comment:
For consistency purpose, return emptyModelValue instead of `undefined` would be better.